metricbeat/module/elasticsearch/ml_job/data.go

```diff
@@ -0,0 +1,67 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package ml_job
+
+import (
+	"encoding/json"
+
+	"github.com/elastic/beats/libbeat/common"
+	s "github.com/elastic/beats/libbeat/common/schema"
+	c "github.com/elastic/beats/libbeat/common/schema/mapstriface"
+	"github.com/elastic/beats/metricbeat/mb"
+)
+
+var (
+	schema = s.Schema{
+		"id":    c.Str("job_id"),
+		"state": c.Str("state"),
+		"data_counts": c.Dict("data_counts", s.Schema{
+			"processed_record_count": c.Int("processed_record_count"),
+		}),
+	}
+)
+
+type jobsStruct struct {
+	Jobs []map[string]interface{} `json:"jobs"`
+}
+
+func eventsMapping(r mb.ReporterV2, content []byte) []error {
```

Comment:
It seems like the return value is being ignored anyway so my following comment my be moot:

Looking at other metricsets in the elasticsearch module, it looks like their `eventsMapping` functions return `error` instead of `[]error`. Further, to build this `error` return object they use `var errs multierror.Errors` to collect multiple errors and then `return `errs.Err()` at the end of the function. For consistency maybe the `eventsMapping` function here should do the same?

Reply:
Yes, need to change it. This changed recently with the schema change from Jaime. Will adapt it accordingly.